src/energy_systems/heat_producers/gas_boiler.jl

```diff
@@ -99,17 +110,50 @@ function check_heat_out(
             unit.output_interfaces[unit.m_heat_out],
             unit.output_interfaces[unit.m_heat_out].target
         )
+        # potential_energy: how much energy needs to be provided directly
         potential_energy_heat_out = exchange.balance + exchange.energy_potential
+        # potential_storage: how much energy can be delivered to storage
         potential_storage_heat_out = exchange.storage_potential
+        # if energy outputed from gas boiler is zero, then return nothing
         if (unit.controller.parameter["load_storages"] ? potential_energy_heat_out + potential_storage_heat_out : potential_energy_heat_out) >= -parameters["epsilon"]
             return (nothing, nothing)
         end
         return (potential_energy_heat_out, potential_storage_heat_out)
-    else
+    else # Provide as much heat as needed for target
         return (-Inf, -Inf)
     end
 end
 
+function calculate_thermal_efficiency(
```

Comment:
In case someone else is wondering about this function, it was agreed that this magic-number implementation, which is based on a study, can serve as a temporary implementation of an efficiency/PLR curve until a more generalised framework for such calculations is implemented.